analyzers/tests/SonarAnalyzer.UnitTest/Rules/TypeParameterNameTest.cs

```diff
@@ -25,8 +28,18 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class TypeParameterNameTest
     {
+        private readonly VerifierBuilder builder = new VerifierBuilder<TypeParameterName>();
```

Comment:
Not really realted to the changes you have done here, but there are test cases missing. Please could you assert the issue message on at least 1 place. And also could you please add a different case where you set the regex parameter to test that the rule configuration is working as expected.